pkg/runtime/authorizer/authorizer.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authorizer
+
+import (
+	"reflect"
+
+	componentsapi "github.com/dapr/dapr/pkg/apis/components/v1alpha1"
+	httpendpointsapi "github.com/dapr/dapr/pkg/apis/httpEndpoint/v1alpha1"
+	"github.com/dapr/dapr/pkg/config"
+	"github.com/dapr/kit/logger"
+)
+
+var log = logger.NewLogger("dapr.runtime.authorizer")
+
+// Type of function that determines if a component is authorized.
+// The function receives the component and must return true if the component is authorized.
+type ComponentAuthorizer func(component componentsapi.Component) bool
+
+// Type of function that determines if an http endpoint is authorized.
+// The function receives the http endpoint and must return true if the http endpoint is authorized.
+type HTTPEndpointAuthorizer func(endpoint httpendpointsapi.HTTPEndpoint) bool
+
+type Options struct {
+	ID           string
+	Namespace    string
+	GlobalConfig *config.Configuration
+}
+
+type Authorizer struct {
+	id        string
+	namespace string
+
+	componentAuthorizers    []ComponentAuthorizer
+	httpEndpointAuthorizers []HTTPEndpointAuthorizer
+}
+
+func New(opts Options) *Authorizer {
+	r := &Authorizer{
+		id:        opts.ID,
+		namespace: opts.Namespace,
+	}
+
+	r.componentAuthorizers = []ComponentAuthorizer{r.namespaceComponentAuthorizer}
+	if opts.GlobalConfig != nil && opts.GlobalConfig.Spec.ComponentsSpec != nil && len(opts.GlobalConfig.Spec.ComponentsSpec.Deny) > 0 {
+		dl := newComponentDenyList(opts.GlobalConfig.Spec.ComponentsSpec.Deny)
+		r.componentAuthorizers = append(r.componentAuthorizers, dl.IsAllowed)
+	}
+
+	r.httpEndpointAuthorizers = []HTTPEndpointAuthorizer{r.namespaceHTTPEndpointAuthorizer}
+
+	return r
+}
+
+func (a *Authorizer) GetAuthorizedObjects(objects interface{}, authorizer func(interface{}) bool) interface{} {
+	reflectValue := reflect.ValueOf(objects)
```

Comment:
Can this be written as a generic method rather than using reflection? This is more complex to read and harder to maintain

Reply:
I didn't author this code, and would rather address this in a followup PR.